### ISSUE TITLE
Training Memory Requirements

### ISSUE THREAD

**Issue by BrianPugh** [xxlong0/ASNDepth#4]
I'm trying to replicate these results, however I'm running into memory limitations with similar hardware reported in the paper.

1. Did you guys use a minibatch of 8 per GPU (effective minibatch size 32 across 4 GPUs) or minibatch size of 2 per GPU (effective minibatch size 8 across 4 GPUs)
2. Did you freeze the backbone or any other sections of the network during training?
3. In your `Depth2normalLight` implementation, did you iterate over random samples, or unroll them over an additional dimension in your tensors?
4. In your `Depth2normalLight` implementation, did you randomly bilinearly sample your guidance featuremap, or did you simply select pixels from the discrete neighbor set?

Thanks!

**Comment by flamehaze1115** [comment]
Thanks for your interests.
1. If use HRNET-48, effective minibatch size 8 across 4 GPUs.
2. We don't freeze any module.
3. Unroll them over an additional dimension in your tensors
4. Simply select pixels from the discrete neighbor set.

**Comment by BrianPugh** [comment]
I've got another question: I see that you guys use 3 channels for your guidance feature, and that you use a sigmoid activation. This gives a max L2 distance of 1.73 between two features. The first part of equation 2 in your paper says you use a normalize gaussian kernel to encode into a range `[0,1]`. However, since the max L2 distance is 1.73, the actual range is only `[0.42, 1]`. Is this correct?

**Comment by BrianPugh** [comment]
also, how/when do you start applying your learning rate decay? Since the backbone/depth-decoder train for longer than the guidance-decoder.

**Comment by BrianPugh** [comment]
do you apply any regularization to the guidance output?

**Comment by flamehaze1115** [comment]
1. We use 3 channels for better visualization. You can adopt more channels here, while in our experiences, 3 channels are enough. The sigma of Gaussian kernel is a hyperparameter, we just simply adopt 0.5 here.
2. After only training depth branch in 20 epochs, we start to train the whole with initial lr=0.00005. You may adjust the learning rate.
3. No regularization to the guidance output.

**Comment by BrianPugh** [comment]
Hmmm, I'm unable to reproduce the results from your paper. However, my guidance maps don't look like yours either. Have you encountered anything like this in your experiments? The estimated surface normal edge also appears noisy instead of crisp.

![individualImage-4](https://user-images.githubusercontent.com/14318576/126725106-71b009bb-1d64-4b48-98d3-7e22d99c10c4.png)
![individualImage-7](https://user-images.githubusercontent.com/14318576/126725195-dde56b05-d5e2-428b-ab57-2e43fd3e7ffc.png)
![individualImage-5](https://user-images.githubusercontent.com/14318576/126725115-4cc8d645-8f3f-434d-9d2b-07888387e315.png)
![individualImage-6](https://user-images.githubusercontent.com/14318576/126725118-6ded5a9a-e503-4a96-9a33-0e61e3fbeb68.png)

**Comment by flamehaze1115** [comment]
Hello. I plan to release the codes in the near future.

**Comment by sysoda** [comment]
Hello.Can you release the training/evaling codes please?

**Comment by phongnhhn92** [comment]
@flamehaze1115 Any plan to release the training code ?
